src/abb_robot_client_RWS2/rws.py

```diff
@@ -0,0 +1,1124 @@
+            
+            return response_json
+        
+        if response_json is None:
+            raise Exception("Robot returning HTTP error " + str(response.status_code))        
+        
+        status = response_json["status"]
+        error_code=int(status["code"])
+        error_message=status.get('msg',"Received error from ABB robot: " + str(error_code))
+        
+        raise ABBException(error_message, error_code)
+
+    def start(self, cycle: Optional[str]='asis',tasks: Optional[List[str]]=['T_ROB1']):#
+        """
+        Start one or more RAPID tasks
+
+        :param cycle: The cycle mode of the robot. Can be `asis`, `once`, or `forever`.
+        :param tasks: One or more tasks to start.
+        """
+
+        rob_tasks = self.get_tasks()
+        for t in tasks:
+            if not t in rob_tasks:
+                raise Exception(f"Cannot start unknown task {t}")
+
+        for rob_task in rob_tasks.values():
+            if not rob_task.motiontask:
+                continue
+            if rob_task.name in tasks:
+                if not rob_task.active:
+                    self.activate_task(rob_task.name)
+            else:
+                if rob_task.active:
+                    self.deactivate_task(rob_task.name)
+
+        payload={"regain": "continue", "execmode": "continue" , "cycle": cycle, "condition": "none", "stopatbp": "disabled", "alltaskbytsp": "true"}
+        res=self._do_post("rw/rapid/execution/start", payload)
+
+    def activate_task(self, task: str):
+        """
+        Activate a RAPID task
+
+        :param task: The name of the task to activate
+        """
+        payload={}
+        self._do_post(f"rw/rapid/tasks/{task}/activate",payload)
+
+    def deactivate_task(self, task: str) -> None:
+        """
+        Deactivate a RAPID task
+
+        :param task: The name of the task to activate
+        """
+        payload={}
+        self._do_post(f"rw/rapid/tasks/{task}/deactivate",payload)
+
+    def stop(self):#
+        """
+        Stop RAPID execution of normal tasks
+        """
+        payload={"stopmode": "stop"}
+        res=self._do_post("rw/rapid/execution/stop", payload)
+
+    def resetpp(self):
+        """
+        Reset RAPID program pointer to main in normal tasks
+        """
+        res=self._do_post("rw/rapid/execution/resetpp") 
+
+    def get_ramdisk_path(self) -> str: #TODO: I have know idea what this is for or how to replicate it
+        """
+        Get the path of the RAMDISK variable on the controller
+
+        :return: The RAMDISK path
+        """
+        res_json = self._do_get("ctrl/$RAMDISK")
+        return res_json["state"][0]["value"]
+
+    def get_execution_state(self) -> RAPIDExecutionState: #
+        """
+        Get the RAPID execution state
+
+        :return: The RAPID execution state
+        """
+        res_json = self._do_get("rw/rapid/execution")
+        print(res_json)

```

Comment:
Did we want to print the results of this? Or was this left behind for testing purposes?